src/unix/fs.c

```
@@ -739,6 +734,45 @@ static int uv__fs_fstat(int fd, uv_stat_t *buf) {
 }
 
 
+typedef ssize_t (*uv__fs_buf_iter_yield)(uv_fs_t* req);
+static ssize_t uv__fs_buf_iter(uv_fs_t* req, uv__fs_buf_iter_yield yield) {
+  unsigned int iovmax;
+  unsigned int nbufs;
+  uv_buf_t* bufs;
+  ssize_t total;
+  ssize_t result;
+
+  iovmax = uv__getiovmax();
+  nbufs = req->nbufs;
+  bufs = req->bufs;
+  total = 0;
+
+  while (nbufs) {
+    req->nbufs = nbufs;
+    if (req->nbufs > iovmax) {
+      req->nbufs = iovmax;
+    }
+
+    result = yield(req);
+    if (result <= 0) {
+      total = result;
```

After mulling it over, I think it's best to ignore the error and indicate a partial write if `total > 0`.  That way data doesn't get lost and if the error isn't temporal, subsequent reads or writes will signal it.